src/basic/sysctl-util.c

```diff
@@ -58,6 +87,12 @@ int sysctl_write(const char *property, const char *value) {
 
         log_debug("Setting '%s' to '%s'", p, value);
 
+        if (shadow) {
+                r = shadow_update(shadow, p, value);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
with the suggestion above you could then drop the conditionalization here.

i think that would be a tiny bit nicer, since it abstracts the difference between the shadow and non-shadow case away.